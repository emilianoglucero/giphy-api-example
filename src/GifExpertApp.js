import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategories] = useState( defaultCategories );

    return(
      <>
      <h2>My GIPHY api example with React ðŸ¥° </h2>
      <hr />
      <AddCategory 
        setCategories={ setCategories }
      />

      {/* <button onClick={ handleAdd }>Agregar</button> */}

      <ol>
        {
          categories.map( category => (
           <GifGrid 
             key={ category }
             category={ category }
           />
          ))
        }
      </ol>


      </>
    )
  } 