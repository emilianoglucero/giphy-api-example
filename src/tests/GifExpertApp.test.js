import React from 'react';
import '../setupTests';

import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('pruebas en el <GifExpertApp />', ()=> {
    

    test('debe de mostrarse correctamente', () => {


        let wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

        
    });

    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Goku'];
        let wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe( categories.length );


        
    })
    
    
    

    
})