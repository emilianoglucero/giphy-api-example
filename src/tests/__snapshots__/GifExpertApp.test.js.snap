// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    My GIPHY api example with React ðŸ¥° 
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en el <GifExpertApp /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    My GIPHY api example with React ðŸ¥° 
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
